pkg/apiserver/rest/dualwriter_mode2.go

```diff
@@ -149,3 +149,26 @@ func enrichObject(orig, copy runtime.Object) (runtime.Object, error) {
 
 	return copy, nil
 }
+
+func (d *DualWriterMode2) Delete(ctx context.Context, name string, deleteValidation rest.ValidateObjectFunc, options *metav1.DeleteOptions) (runtime.Object, bool, error) {
+	legacy, ok := d.Legacy.(rest.GracefulDeleter)
+	if !ok {
+		return nil, false, errDualWriterDeleterMissing
+	}
+
+	deletedLS, async, err := legacy.Delete(ctx, name, deleteValidation, options)
+	if err != nil {
```

Comment:
We're going to have to examine this more closely when we look at failure modes for maintaining data consistency. In the meantime, I was thinking it might be preferable to return right away if the legacy deletion failed. Same thing for mode 3--return right away if Storage deletion failed. It helps keep the two storages more in sync. What do you think? I'm fine keeping it like this for now because we have to revisit the failure modes anyways.

Reply:
agree with the part of failing if deleting an existing object failed in the legacy storage. But if the object doesn't exist already, I think we should also try to have it deleted in US, just in case the object doesn't exist in legacy but still exists in US. Wdyt?

Reply:
Yup, trying the deletion ins Storage if the objects was not found in Legacy makes sense to me as well 👍